shared/src/commonMain/kotlin/org/hyperskill/app/profile/presentation/ProfileActionDispatcher.kt

```diff
@@ -143,13 +141,17 @@ class ProfileActionDispatcher(
                 val streakFreezeProduct = streakFreezeProductResult.await().getOrNull()
                 val badges = badgesDeferred.await().getOrThrow()
 
+                val dailyLearningNotificationHour =
+                    currentProfile.dailyLearningNotificationHour
+                        ?: notificationInteractor.getDailyStudyRemindersIntervalStartHour()
+
                 Message.ProfileFetchResult.Success(
                     profile = currentProfile,
                     streak = streak,
                     streakFreezeState = getStreakFreezeState(streakFreezeProduct, streak),
                     dailyStudyRemindersState = ProfileFeature.DailyStudyRemindersState(
-                        isEnabled = notificationInteractor.isDailyStudyRemindersEnabled(),
-                        startHour = notificationInteractor.getDailyStudyRemindersIntervalStartHour()
+                        isEnabled = currentProfile.isDailyLearningNotificationEnabled,
```

Comment:
if we use profile field, should we remove `NotificationInteractor::isDailyStudyRemindersEnabled` method at all? I think there should only one place to check that daily study reminders are enabled

Also i think that it would be nice to add changing this state in reducing `Message.ProfileChanged`